### PR TITLE
Introduce text2vec-gpt4all module

### PR DESCRIPTION
**Pull request by antas-marcin**
### What's being changed:

This PR introduces `text2vec-gpt4all` module.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/text2vec-gpt4all/neartext/param.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+type ObjectMove struct {
+	ID     string
+	Beacon string
+}
+
+func (g *GraphQLArgumentsProvider) validateNearTextFn(param interface{}) error {
+	nearText, ok := param.(*NearTextParams)
+	if !ok {
+		return errors.New("'nearText' invalid parameter")
```
Comment (parkerduckworth) [inline modules/text2vec-gpt4all/neartext/param.go:57]: Maybe include the invalid param in err msg

File: modules/text2vec-gpt4all/vectorizer/objects.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		client: client,
+	}
+}
+
+type Client interface {
+	Vectorize(ctx context.Context, text string) (*ent.VectorizationResult, error)
+}
+
+// IndexCheck returns whether a property of a class should be indexed
+type ClassSettings interface {
```
Comment (parkerduckworth) [inline modules/text2vec-gpt4all/vectorizer/objects.go:41]: I think this comment belongs somewhere else

File: modules/text2vec-gpt4all/vectorizer/todo_move_out_of_here_movements.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+// CombineVectors combines all of the vector into sum of their parts
+func (v *Vectorizer) CombineVectors(vectors [][]float32) []float32 {
+	return libvectorizer.CombineVectors(vectors)
+}
+
+// MoveTo moves one vector toward another
+func (v *Vectorizer) MoveTo(source []float32, target []float32, weight float32,
+) ([]float32, error) {
+	multiplier := float32(0.5)
```
Comment (parkerduckworth) [inline modules/text2vec-gpt4all/vectorizer/todo_move_out_of_here_movements.go:28]: Where do we get the value `0.5`?

**Review by parkerduckworth — APPROVED** [review]

File: modules/text2vec-gpt4all/vectorizer/todo_move_out_of_here_movements.go (hunk 3 above)
Comment (antas-marcin) [reply modules/text2vec-gpt4all/vectorizer/todo_move_out_of_here_movements.go:28]: This is the original implementation of `nearText` that is used by all `text2vec` modules it is being used to compute the result vector using this formula:
```
out[i] = sourceItem + weight*multiplier*(sourceItem-target[i])
```

File: modules/text2vec-gpt4all/vectorizer/objects.go (hunk 2 above)
Comment (antas-marcin) [reply modules/text2vec-gpt4all/vectorizer/objects.go:41]: removed comment

File: modules/text2vec-gpt4all/additional/projector/projector.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+func (f *FeatureProjector) AdditionalPropertyFn(ctx context.Context,
+	in []search.Result, params interface{}, limit *int,
+	argumentModuleParams map[string]interface{}, cfg moduletools.ClassConfig,
+) ([]search.Result, error) {
+	if parameters, ok := params.(*Params); ok {
+		return f.Reduce(in, parameters)
+	}
+	return nil, errors.New("unknown params")
```
[the remark this answers is not in the document]
Comment (antas-marcin) [reply modules/text2vec-gpt4all/additional/projector/projector.go:53]: I will also change this behaviour during the refactor of module components

File: modules/text2vec-gpt4all/vectorizer/todo_move_out_of_here_movements.go (hunk 3 above)
Comment (parkerduckworth) [reply modules/text2vec-gpt4all/vectorizer/todo_move_out_of_here_movements.go:28]: Ok, interesting. Thanks

File: modules/text2vec-gpt4all/neartext/param.go (hunk 1 above)
Comment (parkerduckworth) [reply modules/text2vec-gpt4all/neartext/param.go:57]: Works for me

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**